specs/template.md

```diff
@@ -0,0 +1,117 @@
+# Spec Title
+
+## Overview
+
+*The name of the component, along with a high-level description.*
+
+### Background
+
+*Relevant historical or background information, related existing issues, etc.*
+
+### Use Cases
+
+*Primary use cases for this component.*
+  
+### Features
+
+*A list of the key features unique to this component.*
+
+### Risks and Challenges
+
+*Notable risks or challenges associated with implementing the component. Would we need to make any breaking changes to other parts of fast-dna in order to achieve this component's goals?*
+
+### Prior Art/Examples
+
+*Screenshots and/or links to existing, canonical or exemplary implementations of the component.*
+
+---
+
+## Design
+
+*Describe the design of the component, thinking through several perspectives:*
+
+- *A customer using the component on a web page.*
+- *A developer building an app with the component and interacting through HTML/CSS/JS.*
+- *A designer customizing the component.*
+
+### API
+
+*The key elements of the component's public API surface:*
+
+- *Component Name*
+- *Props/Attrs*
+- *Methods*
+- *Events*
+
+*Consider high and low-level APIs. Attempt to design a powerful and extensible low-level API with a higher-level API for developer/designer ergonomics and simplicity.*
+
+### Anatomy and Appearance
+
+*Screenshots and/or description of the basic appearance of the component. Outline its structure with a diagram of its visual tree (shadow dom). Enumerate key areas of visual customization, such as:*
+
+- *Slot Names*
+- *Host Classes*
+- *Slotted Content/Slotted Classes*
+- *CSS Parts*
+
+---
+
+## Implementation
+
+*Important aspects of the planned implementation, with careful consideration of web standards and integration.*
+
+### States
+
+*Key component states, valid state transitions, and how interactions trigger a state transition.*
+
+### Accessibility
+
+*Consider the accessibility of the component, including:*
+
+- *Use with Assistive Technology*
```

Comment:
I wonder if an example might be helpful here. For instance, I don't think that we should build for specific AT solutions (NVDA, Narrator, Voice Over, JAWS, etc) - and I don't think this is what's being proposed, it's just how I'm reading it. Perhaps detailing that working in the shadow dom has implications on how roles are presented to AT or something?

Reply:
Added a quick note based on your recommendation.